standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdatePart.java

```diff
@@ -484,14 +477,14 @@ public Map<String, Map<String, String>> updatePartitionColumnStatistics(Map<Stri
                                                       String validWriteIds, long writeId,
                                                       List<TransactionalMetaStoreEventListener> transactionalListeners)
           throws MetaException {
+
     JDOConnection jdoConn = null;
-    Connection dbConn = null;
-    boolean committed = false;
+    Transaction tx = pm.currentTransaction();
```

Comment:
why do we need txn here?

Reply:
It's used to mark we have opened a transaction explicitly, when `jdoConn = pm.getDataStoreConnection()`, it will get the connection from the primary connection pool, https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/PersistenceManagerProvider.java#L307

https://github.com/datanucleus/datanucleus-core/blob/master/src/main/java/org/datanucleus/store/AbstractStoreManager.java#L373

https://github.com/datanucleus/datanucleus-core/blob/master/src/main/java/org/datanucleus/store/connection/ConnectionManager.java#L92